be/src/olap/generic_iterators.cpp

```diff
@@ -116,13 +120,91 @@ class MergeIteratorContext {
     MergeIteratorContext(RowwiseIterator* iter, std::shared_ptr<MemTracker> parent)
             : _iter(iter), _block(iter->schema(), 1024, std::move(parent)) {}
 
+    MergeIteratorContext(const MergeIteratorContext&) = delete;
+    MergeIteratorContext(MergeIteratorContext&&) = delete;
+    MergeIteratorContext& operator=(const MergeIteratorContext&) = delete;
+    MergeIteratorContext& operator=(MergeIteratorContext&&) = delete;
+
     ~MergeIteratorContext() {
         delete _iter;
         _iter = nullptr;
     }
 
     // Initialize this context and will prepare data for current_row()
     Status init(const StorageReadOptions& opts);
+    Status v_init(const StorageReadOptions& opts);
+
+    bool is_null(vectorized::ColumnPtr cp) const {
+        return cp->is_nullable() && cp->is_null_at(_v_index_in_block);
+    }
+
+    int compare_cell(vectorized::ColumnPtr l_cp, vectorized::ColumnPtr r_cp) const {
```

Comment:
这里不应该用Filed去进行比较，注释里写的很明确了。   
  /// Returns value of n-th element in universal Field representation.
 /// Is used in rare cases, since creation of Field instance is expensive usually.

本身Column提供了Compare_at的接口，应该基于那个实现，并且也处理了null的逻辑。参考SortCursor的实现代码